evm/evm.go

```diff
@@ -0,0 +1,36 @@
+package evm
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+)
+
+var (
+	// MonomerGenesisRootHash is the known root hash of the monomer ethereum state trie after all predeployed contracts are created.
+	MonomerGenesisRootHash = common.HexToHash("0x4e171626bc7f95d0b053dab23c87327ee0266dda88b213a3e1f1357b741c0c35")
+	// TODO: is there a specific address we want for MonomerEVMOriginAddress?
+	// MonomerEVMOriginAddress is the address used for executing transactions in the monomer EVM.
+	MonomerEVMOriginAddress = common.HexToAddress("0xae76d9126c32d7eafe3f5b7bd5a7b44f2d5bb8b1")
+)
+
+func NewEVM(ethState vm.StateDB) (*vm.EVM, error) {
+	return vm.NewEVM(
+		vm.BlockContext{
+			BlockNumber: big.NewInt(0),
```

Comment:
Yeah that's a good idea. It will impact how certain opcodes function but the block number or time where those upgrades take effect are configured through `params.ChainConfig`